src/profile/models/profile.rs

```diff
@@ -0,0 +1,903 @@
+use std::{
+    collections::HashMap,
+    fmt,
+};
+
+use serde::{
+    de::{
+        self,
+        Visitor,
+    },
+    Deserialize,
+    Deserializer,
+    Serialize,
+};
+use serde_json::Value;
+use serde_repr::{
+    Deserialize_repr,
+    Serialize_repr,
+};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "PascalCase")]
+pub(crate) struct ProfilePayload {
+    pub(crate) results: Vec<Profile>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "PascalCase")]
+pub struct Profile {
+    #[serde(deserialize_with = "deserialize_oid")]
+    /// account_id
+    pub account_id: String,
+
+    /// display_name
+    pub display_name: PlatformName,
+
+    /// platform_names
+    pub platform_names: Option<Vec<PlatformName>>,
+
+    /// player_level
+    pub player_level: u8,
+
+    /// load_out_preset
+    pub load_out_preset: Option<LoadOutPreset>,
+
+    /// load_out_inventory
+    pub load_out_inventory: LoadOutInventory,
+
+    #[serde(deserialize_with = "deserialize_oid_or_none")]
+    /// guild_id
+    pub guild_id: Option<String>,
+
+    /// guild_name
+    pub guild_name: Option<String>,
+
+    /// guild_tier
+    pub guild_tier: Option<GuildTier>,
+
+    #[serde(rename = "GuildXP")]
+    /// guild_xp
+    pub guild_xp: Option<i32>,
+
+    /// guild_class
+    pub guild_class: Option<u8>,
+
+    /// guild_emblem
+    pub guild_emblem: Option<bool>,
+
+    #[serde(deserialize_with = "deserialize_oid_or_none")]
+    /// alliance_id
+    pub alliance_id: Option<String>,
+
+    /// player_skills
+    pub player_skills: HashMap<PlayerSkill, i32>,
+
+    /// challenge_progress
+    pub challenge_progress: Vec<ChallengeProgress>,
+
+    /// death_marks
+    pub death_marks: Vec<String>,
+
+    /// harvestable
+    pub harvestable: bool,
+
+    /// death_squadable
+    pub death_squadable: bool,
+
+    #[serde(deserialize_with = "deserialize_date")]
+    /// created
+    pub created: i64,
+
+    /// migrated_to_console
+    pub migrated_to_console: bool,
+}
+
+#[derive(Serialize, Debug, Clone)]
+pub struct PlatformName {
+    /// name
+    pub name: String,
+
+    /// platform
+    pub platform: Platform,
+}
+
+impl<'de> Deserialize<'de> for PlatformName {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct SplitFieldVisitor;
+
+        impl<'de> Visitor<'de> for SplitFieldVisitor {
+            type Value = PlatformName;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a string with a byte code for platform at the end")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                // Split string
+                let (display_name, platform_byte) = value.split_at(value.len() - 1);
```

Comment:
Potentially dangerous to do. `str.len()` only looks at the length of bytes, without taking UTF8 into account.

Take for example `"завтра".len()`, which evaluates to `12`.

Do this instead
```suggestion
                let (display_name, platform_byte) = value.split_at(value.chars().count() - 1);
```

Reply:
nevermind, it's the last BYTE. My bad.